monitoring/datadog_event.py

```diff
@@ -106,51 +123,41 @@ def main():
                 choices=['error', 'warning', 'info', 'success']
             ),
             aggregation_key=dict(required=False, default=None),
-            source_type_name=dict(
-                required=False, default='my apps',
-                choices=['nagios', 'hudson', 'jenkins', 'user', 'my apps',
-                         'feed', 'chef', 'puppet', 'git', 'bitbucket', 'fabric',
-                         'capistrano']
-            ),
             validate_certs = dict(default='yes', type='bool'),
         )
     )
 
-    post_event(module)
+    # Prepare Datadog
+    if not HAS_DATADOG:
+        module.fail_json(msg='datadogpy required for this module')
 
-def post_event(module):
-    uri = "https://app.datadoghq.com/api/v1/events?api_key=%s" % module.params['api_key']
+    options = {
+        'api_key': module.params['api_key'],
+        'app_key': module.params['app_key']
+    }
 
-    body = dict(
-        title=module.params['title'],
-        text=module.params['text'],
-        priority=module.params['priority'],
-        alert_type=module.params['alert_type']
-    )
-    if module.params['date_happened'] != None:
-        body['date_happened'] = module.params['date_happened']
-    if module.params['tags'] != None:
-        body['tags'] = module.params['tags']
-    if module.params['aggregation_key'] != None:
-        body['aggregation_key'] = module.params['aggregation_key']
-    if module.params['source_type_name'] != None:
-        body['source_type_name'] = module.params['source_type_name']
+    initialize(**options)
+
+    _post_event(module)
+
+
+def _post_event(module):
+    try:
+        msg = api.Event.create(title=module.params['title'],
+                               text=module.params['text'],
+                               tags=module.params['tags'],
+                               priority=module.params['priority'],
+                               alert_type=module.params['alert_type'],
+                               aggregation_key=module.params['aggregation_key'],
+                               source_type_name='ansible')
+        if msg['status'] != 'ok':
+            module.fail_json(msg=msg)
 
-    json_body = module.jsonify(body)
-    headers = {"Content-Type": "application/json"}
+        module.exit_json(changed=True, msg=msg)
+    except Exception, e:
```

Comment:
Please replace all

~~~python
except Exception, e:
~~~
with

~~~python
except Exception:
     e = get_exception()
~~~
to get a green build.